lib/rustler_precompiled.ex

```diff
@@ -803,6 +810,37 @@ defmodule RustlerPrecompiled do
     end)
   end
 
+  defp max_retries(options) do
+    value = Keyword.get(options, :max_retries, 3)
+
+    if value not in 0..15,
+      do: raise("attempts should be between 0 and 15. Got: #{inspect(value)}")
+
+    value
+  end
+
+  defp with_retry(fun, attempts) when attempts in 0..15 do
+    task = Task.async(fun)
+    first_try = Task.await(task, :infinity)
```

Comment:
you're doing async/await right away. If you don't use Task for the timeout, is it useful?

Reply:
I was being lazy, but yeah, it's not useful :sweat_smile: 
Fixed in d6a323e

Reply:
there's one more similar place :)

Reply:
OMG :sweat: 
https://github.com/philss/rustler_precompiled/commit/94b2b757f4c7eb7f5553f5418a39e47fee2ef897

Thank you!